cpp/src/arrow/python/common.h

```diff
@@ -257,6 +272,15 @@ struct PyBytesView {
       this->size = PyByteArray_GET_SIZE(obj);
       this->ref.reset();
       return Status::OK();
+    } else if (PyMemoryView_Check(obj)) {
+      PyObject* contig_view = PyMemoryView_GetContiguous(obj, PyBUF_READ, 'C');
+      RETURN_IF_PYERROR();
+      this->ref.reset(contig_view);
+      std::shared_ptr<Buffer> buffer;
+      ARROW_ASSIGN_OR_RAISE(buffer, PyBuffer::FromPyObject(contig_view));
```

Comment:
You shouldn't need this, just call `PyMemoryView_GET_BUFFER(obj)` and access the fields on the resulting `Py_buffer` struct.